Source/GameView/Repositories/GameAPIRepo.swift

```diff
@@ -0,0 +1,205 @@
+//
+//  GameAPIRepo.swift
+//  CommonFiles
+//
+//  Created by Tiewhan Smith on 2020/03/06.
+//
+
+import Foundation
+
+///Structure is only used to model JSON respones. Contains the a structure of Apps
+struct AppList: Codable {
+  let applist: Apps
+}
+
+///Structure is used to model JSON respones. Contains the list of games
+struct Apps: Codable {
+  let apps: [JsonGame]
+}
+
+///Structure is used to model JSON respones. This is the actual game
+struct JsonGame: Codable {
+  let appid: Int
+  let name: String
+}
+
+public class GameAPIRepo: GameRepository {
+  
+  private var observers: [String: GameRepositoryObserver] = [:]
+  
+  public init() {
+    
+  }
+  
+  public func getGameList(with serviceCaller: ServiceCaller) {
+
+    let url: URL = URL(string: SteamURLComponents.gameListURL)!
+    
+    // MARK: - Call Failed Closure
+    serviceCaller.callFailed = { error in
+      
+    }
+    
+    // MARK: - ---------------------------
+
+    // MARK: - Call Succeeded Closure
+    serviceCaller.callSucceeded = { [weak self] data, _ in
+
+      guard let self = self else {
+        return
+      }
+
+      let games = self.decodeGames(data: data)
+
+      var smallGameList: [Game] = []
+
+      //This limit exists until sectioning with the table works.
+      let size: Int = games.count >= 25 ? 25 : games.count
+
+      if games.count >= 1 {
+
+        for index in 0...(size - 1) {
+          smallGameList.append(games[index])
+        }
+
+        self.notifyGameListLoaded(gameList: smallGameList)
+
+      }
+
+    }
+    
+    // MARK: - ---------------------------
+    
+    do {
+      try serviceCaller.makeServiceCall(with: url, and: DataBundle())
+    } catch {
+      fatalError("Service call not set up properly")
+    }
+    
+  }
+  
+  public func getGameDetails(of gameList: [Game], with serviceCaller: ServiceCaller) {
+    
+    let dispatchGroup = DispatchGroup()
+    
+    gameList.forEach({ (game) in
+      
+      dispatchGroup.enter()
+      
+      let dataBundle = DataBundle()
+      dataBundle.extraData["game"] = game
+      
+      // MARK: - Call Succeeded Closure
+      serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+        
+        guard let game = dataBundle.extraData["game"] as? Game else {
+          return
+        }
+        
+        guard let self = self else {
+          return
+        }
+        
+       self.decodeGameDetails(using: data, basedOn: game)
```

Comment:
If we only have one function call, we can just call `self?.decodeGameDetails` instead of having to have an unnecessary extra line of unwrapping self